src/js/client/editprefset-settingsTable-widget.js

```diff
@@ -0,0 +1,259 @@
+ * The first time it is created though, we want to set the default values,
+ * and trigger them via change applier so they are set on the parent.
+ *
+ * @param (Object) that
+ */
+gpii.devpmt.settingsTable.filterInit = function (that) {
+    if (that.model.settingsFilter === null) {
+        that.applier.change("settingsFilter", "");
+    }
+    if (that.model.allSettingsEnabled === null) {
+        that.applier.change("allSettingsEnabled", "mysettings");
+    }
+};
+
+gpii.devpmt.settingsTable.enableProductListener = function (that, devpmt, event) {
+    var checked = event.currentTarget.checked;
+    var context = event.currentTarget.dataset.context;
+    var product = event.currentTarget.dataset.product;
+    devpmt.editProductEnabled(checked, context, product);
+};
+
+gpii.devpmt.settingsTable.updateTermUsage = function (that) {
+    var all = 0;
+    var npsetKeys = {};
+    fluid.each(that.model.flatPrefs.contexts, function (context) {
+        fluid.each(context.preferences[that.options.appUri], function (setting, settingKey) {
+            npsetKeys[settingKey] = true;
+        });
+    });
+    fluid.each(that.options.solution.settingsHandlers, function (settingsHandler) {
+        fluid.each(settingsHandler.supportedSettings, function (setting) {
+            if (setting.schema) {
+                all++;
+            }
+        });
+    });
+    that.applier.change("termUsage", {
+        all: all,
+        npset: Object.keys(npsetKeys).length
+    });
+};
+
+/**
+ * Update the currently displayed settings in the table based upon a
+ * filter string.
+ *
+ * @param filterText
+ */
+gpii.devpmt.settingsTable.filterSettings = function (that, settingsRows, lunrIndex, filterText) {
+    if (!lunrIndex) {
+        return;
+    }
+
+    var showBasedOnAllFilter = function (settingId) {
+        var togo = false;
+        if (that.model.allSettingsEnabled === "allsettings") {
+            togo = true;
+        }
+        else {
+            fluid.each(that.model.flatPrefs.contexts, function (context) {
+                fluid.each(context.preferences[that.options.appUri], function (setting, settingKey) {
+                    if (settingKey === settingId) {
+                        togo = true;
+                    }
+                });
+            });
+        }
+        return togo;
+    };
+
+    // TODO Refactor with gpii.devpmt.productList.filterProductList
+    settingsRows.hide();
+
+    var results = gpii.devpmt.lunrListFilterSearch(lunrIndex, filterText);
+
+    // TODO Remove this double loop
+    fluid.each(settingsRows, function (setting) {
+        fluid.each(results, function (result) {
+            if (result.ref === $(setting).data("term")) {
+                if (showBasedOnAllFilter(result.ref)) {
+                    $(setting).show();
+                }
+            }
+        });
+    });
+};
+
+gpii.devpmt.settingsTable.updateLunrIndex = function (that) {
+    that.lunrIndex = lunr(function () {
```

Comment:
"We are lunies, and we are proud!" - https://www.youtube.com/watch?v=3Kcwe2uY6Zc&feature=youtu.be&t=205

Reply:
Intense....